db/seeds.rb

```diff
@@ -8,14 +8,65 @@
 #     MovieGenre.find_or_create_by!(name: genre_name)
 #   end
 
-# Create the default user
-user = User.create_or_find_by!(
-  first_name: "Jeremy", last_name: "Caley", email: "jeremy",
-  password: "password", password_confirmation: "password"
-)
+require 'faker'
 
-user.workspaces.create_or_find_by!(name: "Caley", owner_id: user.id)
-puts "Workspace created: #{user.workspaces.first.name}"
+email_providers = ['gmail.com', 'yahoo.com', 'hotmail.com', 'outlook.com', 'aol.com']
 
-user.teams.create_or_find_by!(name: "Caley", workspace_id: user.workspaces.first.id)
-puts "Team created: #{user.teams.first.name}"
+# Create multiple users
+5.times do
```

Comment:
This all looks great! Would you mind adding a few `messages` as well? The `messages` table represents actual emails.

Reply:
Thanks for the feedback. I have included messages and pushed the updated code. Let me know if you need any other changes.

Just also wanted to give heads up when we try to login with populated data we encounter issue with below code

  ```
def establish_imap_connection
    @imap_connection ||= Net::IMAP.new(self.imap_server, port: self.imap_port, ssl: self.imap_ssl)
    @imap_connection.login(self.email, self.password)
    @imap_connection
  rescue Net::IMAP::Error => error
    puts "Failed to connect to IMAP: #{error.message}"
    nil
  end
```
As it tries to establish connection with populated imap_server -  Net::OpenTimeout (Timeout to open TCP connection to imap.caley.to:993

Reply:
Ah, yeah that should be fixed soon! Thank you for catching it. @al3rez let's take note of this.